### PR TITLE
ENH: Add a force argument to `numpy()`

### LINKED ISSUE #20778: Support One step transform to numpy
## 🚀 Feature
Writing         `print tensor.detach().cpu().numpy()` gets really annoying. Can you provide a method that just takes you to numpy in one step?

Like `fnumpy` (force to numpy). So if it has gradient, it will detach it for you, if it is cuda, it will move to CPU for you etc.

### PR DESCRIPTION
**Pull request by HaoZeke**
**Reopened** to help with merge issues. See #59790 for full context.

Fixes #20778. Helps #71688.

Finalizes @martinPasen's force argument for `Tensor.numpy()`. It is set to False by default. If it's set to True then we:
1. detatch the Tensor, if requires_grad == True
2. move to cpu, if not on cpu already
3. Uses .resolve_conj() if .is_conj() == True
4. Uses .resolve_neg() if .is_neg() == True

cc @albanD

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Looks good!
Thanks for the new PR.

**Comment by vadimkantorov** [comment]
In theory, could it / numpy() in general support more sparse tensor types (those that are supported by numpy of course)?

**Comment by HaoZeke** [comment]
@janeyx99 could you look into the PyTorchBot issue?

**Comment by albanD** [comment]
Ho I didn't realize the bot didn't merge that sorry!

**Comment by janeyx99** [comment]
> @janeyx99 could you look into the PyTorchBot issue?

Ah! We've migrated off the natural language commands and moved to a more CLI format--please check our wiki for the valid commands now: https://github.com/pytorch/pytorch/wiki/Bot-commands

**Comment by vadimkantorov** [comment]
@alband Can we allow automatic detach / force = True when doing automatic numpy conversion?

E.g. when visualizing Torch tensors with matplotlib: `plt.plot(x, y.requires_grad_())` will fail because y is not detached, but it's quite useful for visualization purposes.

**Comment by albanD** [comment]
Do you have specific examples in mind?
The plotting code should be able to use force() explicitly there so that the end user doesn't have to specify it?

**Comment by vadimkantorov** [comment]
My treatment is: the better and better Pytorch --> NumPy interop becomes, the more value is in doing detach (or even gpu->cpu) automatically (if want to protect, can add a readonly bit in NumPy tensors).

At least for PyTorch->matplotlib quick vis the explicit `.numpy()` calls do make the code less readable.

Device transfer is a bit more understandable, but why can't attached-to-graph tensors be passed for zero-copy NumPy conversion?
